### PR TITLE
Pseudo labeling

### LINKED ISSUE #127: Allow for f-strings in logging?
Pylint complains about it right now, but it's a lot more readable. Does it make sense to disable this pylint error and use f-strings?

### LINKED ISSUE #140: Add sparse storage to NOTELA

### LINKED ISSUE #141: Data augmentations/mixup for pseudolabeling
Should we use data augmentations/ mixup for finetuning on pseudolabels? I think data augs should be significantly less aggressive for the pseudolabeling. However, do we want that to mean a different set of augs, weaker augs, no augs, or something else entirely?

### LINKED ISSUE #142: Cross Entropy Loss Extremely High
On pseudo branch, cross entropy loss is very high, ~200 on par with BCEWL, but typical is expected to be about 4

### LINKED ISSUE #143: Program crashes if cfg.class_list empty

### LINKED ISSUE #144: Dataloader does not raise error if files exist but loading fails
Filtering out files that failed to load happens after checking if there are valid files, but the order should be the other way around

### PR DESCRIPTION
**Pull request by benjamin-cates**

### CODE REVIEW

**Comment by benjamin-cates** [comment]
Something in this branch is causing logger statements to be printed twice and I'm not sure why. It doesn't happen on main 😭

**Comment by benjamin-cates** [comment]
The most recent commit has a fix for the problem. Calling `config.Config()` causes the logger to duplicate 😭 . Should prolly store that in the spaghetti code tupperware for later.

**Comment by benjamin-cates** [comment]
FIXED: Note for later: breaks if class list is None

**Comment by benjamin-cates** [comment]
Ok one issue I have with this is that if you're just running on pseudo labels it loads the training dataset multiple times even though it's not being used. We need a command line argument or something to specify the kind of fine tuning like "pseudo", "nutella" or "none"

**Review by marco-bazzani — COMMENTED** [review]
Fixes #141

**Comment by benjamin-cates** [comment]
Sean gonna love this pull request 🥰

File: pyha_analyzer/default_config.yml [hunk 1]
```diff
@@ -40,7 +44,7 @@ chunk_margin_s: 0.4 # Margin on either side of annotation
 logging: true
 logging_freq: 20 # Batches per log
 wandb_entity: "acoustic-species-identification"
-wandb_project: "acoustic-species-reu2023"
+wandb_project: "nutella"
```
Comment (benjamin-cates) [inline pyha_analyzer/default_config.yml]: oop shouldn't change default config here

File: .pylintrc [hunk 2]
```diff
@@ -414,14 +414,15 @@ confidence=HIGH,
...
+	logging-fstring-interpolation,
         bad-inline-option,
         locally-disabled,
         file-ignored,
         suppressed-message,
         useless-suppression,
         deprecated-pragma,
         use-symbolic-message-instead,
-        trailing-whitespace
+	trailing-whitespace
```
Comment (benjamin-cates) [inline .pylintrc:425]: Thank god 🥰

Comment (marco-bazzani) [reply .pylintrc:425]: lmao

File: pyha_analyzer/default_config.yml (hunk 1 above)
Comment (marco-bazzani) [reply pyha_analyzer/default_config.yml]: good catch

File: pyha_analyzer/chunking_methods/csv_cleaner.py
Comment (sprestrelski) [inline pyha_analyzer/chunking_methods/csv_cleaner.py:1]: why is this file being edited in this PR

File: pyha_analyzer/get_test_map.py
Comment (sprestrelski) [inline pyha_analyzer/get_test_map.py:1]: why was this deleted? also seems outside scope of PR

**Review by sprestrelski — COMMENTED** [review]
there are so many restructures in this PR... I can't wait to test it later

File: pyha_analyzer/get_test_map.py
Comment (marco-bazzani) [reply pyha_analyzer/get_test_map.py:1]: Talked to sean, said we weren't using it and was fine to delete

File: pyha_analyzer/chunking_methods/csv_cleaner.py
Comment (marco-bazzani) [reply pyha_analyzer/chunking_methods/csv_cleaner.py:1]: Just removed some trailing whitespace

**Comment by Sean1572** [comment]
I'm concerned that this maybe too big, or may conflict with notela changes as well
